chain/consensus/mir/mempool/handlers/batches.go

```diff
@@ -21,54 +21,37 @@ func IncludeBatchCreation(
 	params *types.ModuleParams,
 	commonState *types.State,
 ) {
-	state := &State{
-		State:    commonState,
-		NewTxIDs: nil,
-	}
-
-	dsl.UponNewRequests(m, func(txs []*requestpb.Request) error {
-		mpdsl.RequestTransactionIDs(m, mc.Self, txs, &requestTxIDsContext{txs})
-		return nil
-	})
-
 	mpdsl.UponTransactionIDsResponse(m, func(txIDs []t.TxID, context *requestTxIDsContext) error {
-		for i := range txIDs {
-			state.TxByID[txIDs[i]] = context.txs[i]
-		}
-		state.NewTxIDs = append(state.NewTxIDs, txIDs...)
-		return nil
-	})
-
-	mpdsl.UponRequestBatch(m, func(origin *mempoolpb.RequestBatchOrigin) error {
-		var txIDs []t.TxID
 		var txs []*requestpb.Request
-		batchSize := 0
-
 		txCount := 0
-		for _, txID := range state.NewTxIDs {
-			tx := state.TxByID[txID]
+		for i := range txIDs {
+			tx := context.txs[i]
+			commonState.TxByID[txIDs[i]] = tx
```

Comment:
Is there a reason why we still save the transaction in the commonState buffer? It would be nice if this buffer didn't exist at all and the transactions were directly added in the new batch and immediately forgotten.

Reply:
Fixed